Framework/API/inc/MantidAPI/ISpectrum.h

```diff
@@ -115,10 +98,117 @@ class DLLExport ISpectrum {
 
   void setSpectrumNo(specnum_t num);
 
-  virtual bool hasDx() const;
-  virtual void resetHasDx();
+  bool hasDx() const;
+  void resetHasDx();
+
+  virtual HistogramData::Histogram histogram() const { return histogramRef(); }
+  template <typename... T> void setHistogram(T &&... data) {
+    HistogramData::Histogram histogram(std::forward<T>(data)...);
+    checkHistogram(histogram);
+    mutableHistogramRef() = std::move(histogram);
+  }
+
+  HistogramData::BinEdges binEdges() const { return histogramRef().binEdges(); }
+  HistogramData::BinEdgeStandardDeviations binEdgeStandardDeviations() const {
+    return histogramRef().binEdgeStandardDeviations();
+  }
+  HistogramData::Points points() const { return histogramRef().binEdges(); }
```

Comment:
Is there a copy-paste error here? `histogramRef().binEdges();`. If `Histogram` is in `point` mode, we convert to bin edges and then back to points?